File: android/src/main/kotlin/com/example/design_engine/layer2_core/light/EOnTertiaryContainerLight.kt
package com.example.design_engine.layer2_core.light
import com.example.design_engine.R
import com.example.design_engine.layer3_logic.AppTheme
object EOnTertiaryContainerLight {
    fun getResId(activeTheme: AppTheme): Int {
        return when (activeTheme) {
            AppTheme.SYSTEM -> R.string.csyl36
            AppTheme.STANDARD -> R.color.cstl36
            AppTheme.BLUE -> R.color.cbl36
            AppTheme.GREEN -> R.color.cgl36
            AppTheme.RED -> R.color.crl36
            AppTheme.NOTHING -> R.color.cnol36
        }
    }
}

File: lib/layer2_core/light/e_on_tertiary_container_light.dart
import '../../layer1_raw/design_token.dart';
import '../../layer1_raw/light/system.dart';
import '../../layer1_raw/light/standard.dart';
import '../../layer1_raw/light/blue.dart';
import '../../layer1_raw/light/green.dart';
import '../../layer1_raw/light/red.dart';
import '../../layer1_raw/light/nothing.dart';

class EOnTertiaryContainerLight {
  static DesignToken getToken(dynamic activeTheme) {
    switch (activeTheme.toString().split('.').last) {
      case 'system': return csyl36;
      case 'standard': return cstl36;
      case 'blue': return cbl36;
      case 'green': return cgl36;
      case 'red': return crl36;
      case 'nothing': return cnol36;
      default: return cstl36;
    }
  }
}
